Mathlib/SetTheory/Ordinal/Arithmetic.lean

```diff
@@ -2443,6 +2443,30 @@ theorem sup_mul_nat (o : Ordinal) : (sup fun n : ℕ => o * n) = o * ω := by
     exact sup_eq_zero_iff.2 fun n => zero_mul (n : Ordinal)
   · exact (mul_isNormal ho).apply_omega0
 
+/-- The order isomorphism between ℕ and the first ω ordinals. -/
+@[simps! apply]
+def relIso_nat_omega0 : ℕ ≃o Iio ω where
+  toFun n := ⟨n, nat_lt_omega0 n⟩
+  invFun n := Classical.choose (lt_omega0.1 n.2)
+  left_inv n := by
+    have h : ∃ m : ℕ, n = (m : Ordinal) := ⟨n, rfl⟩
+    exact (Nat.cast_inj.1 (Classical.choose_spec h)).symm
+  right_inv n := Subtype.eq (Classical.choose_spec (lt_omega0.1 n.2)).symm
+  map_rel_iff' := by simp
+
+theorem relIso_nat_omega0_coe_symm_apply (o : Iio ω) : relIso_nat_omega0.symm o = o.1 := by
```

Comment:
This should be a `simp` lemma (and then the last two lines of `relIso_nat_omega0_coe_symm_apply` can be replaced by simply `simp`).

Reply:
Did you mean the last lines of the monotone theorem(s)? Either way `simp` doesn't `ext` here so it's still needed.

Reply:
Yeah, my bad.